src/ddmd/astbase.d

```diff
@@ -5888,6 +5920,31 @@ struct ASTBase
         }
     }
 
+    extern (C++) final class StaticForeach : RootObject
+    {
+        Loc loc;
+
+        ForeachStatement aggrfe;
+        ForeachRangeStatement rangefe;
+
+        final extern (D) this(Loc loc,ForeachStatement aggrfe,ForeachRangeStatement rangefe)
```

Comment:
I tried to follow the sparse style of the surrounding code, but failed in some places. I'll change it.
(This really ought to be done by a tool, this wastes so much time.)